docs/source/en/api/pipelines/stable_diffusion/pix2pix_zero.mdx

```diff
@@ -93,9 +94,75 @@ images = pipeline(
 images[0].save("edited_image_dog.png")
 ```
 
-**Based on an input image**
+### Based on an input image
 
-_Coming soon_
+When the pipeline is conditioned on an input image, we first obtain an inverted
+noise from it using a `DDIMInverseScheduler` with the help of a generated caption. Then 
+the inverted noise is used to start the generation process. 
+
+First, let's load our pipeline: 
+
+```py
+import torch
+from transformers import BlipForConditionalGeneration, BlipProcessor
+from diffusers import DDIMScheduler, DDIMInverseScheduler, StableDiffusionPix2PixZeroPipeline
+
+captioner_id = "Salesforce/blip-image-captioning-base"
+processor = BlipProcessor.from_pretrained(captioner_id)
+model = BlipForConditionalGeneration.from_pretrained(captioner_id, torch_dtype=torch.float16, low_cpu_mem_usage=True)
+
+sd_model_ckpt = "CompVis/stable-diffusion-v1-4"
+pipeline = StableDiffusionPix2PixZeroPipeline.from_pretrained(
+    sd_model_ckpt,
+    caption_generator=model,
+    caption_processor=processor,
+    torch_dtype=torch.float16,
+    safety_checker=None,
+)
+pipeline.scheduler = DDIMScheduler.from_config(pipeline.scheduler.config)
+pipeline.inverse_scheduler = DDIMInverseScheduler.from_config(pipeline.scheduler.config)
+pipeline.enable_model_cpu_offload()
+```
+
+Then, we load an input image for conditioning and obtain a suitable caption for it: 
+
+```py
+import requests
+from PIL import Image
+
+img_url = "https://github.com/pix2pixzero/pix2pix-zero/raw/main/assets/test_images/cats/cat_6.png"
+raw_image = Image.open(requests.get(img_url, stream=True).raw).convert("RGB").resize((512, 512))
+caption = pipeline.generate_caption(raw_image)
+```
+
+Then we employ the generated caption and the input image to get the inverted noise: 
+
+```py 
+inv_latents, inv_image = pipeline.invert(caption, image=raw_image)
+```
+
+Now, generate the image with edit directions: 
+
+```py
+source_prompts = 4 * ["a cat sitting on the street", "a cat playing in the field", "a face of a cat"]
```

Comment:
Can we maybe also use Flan-T5 here to generate? No need to test it, happy to test it after :-)

Reply:
Added a comment to emphasize that generation can be automated with Flan. See the updates. 

I think it's slightly better because otherwise, the example code snippet becomes a bit too longer, no?